integrations/GA/browser.js

```diff
@@ -29,88 +56,638 @@ class GA {
+              category: item.category || props.category,
+              list: props.list_id || props.category || "search results",
+              brand: props.brand,
+              variant: filters + "::" + sorts,
+              price: item.price,
+              position: getProductPosition(item, products),
+            };
+
+            impressionObj = extend(
+              impressionObj,
+              metrics(
+                item.properties,
+                dimensionsArray,
+                metricsArray,
+                contentGroupingsArray
+              )
+            );
+
+            for (var prop in impressionObj) {
+              if (impressionObj[prop] === undefined) delete impressionObj[prop];
+            }
+            ga("ec:addImpression", impressionObj);
+          });
+          pushEnhancedEcommerce(rudderElement, this.inputs);
+          break;
+        default:
+          var contextOpts; //need to implement
+          var interfaceOpts = this.inputs;
+          var opts = defaults(options || {}, contextOpts);
+          opts = defaults(opts, interfaceOpts);
+
+          var eventCategory = rudderElement.message.properties.category;
+          var eventAction = rudderElement.message.event;
+          var eventLabel = rudderElement.message.properties.label;
+          var eventValue = "";
+          if (rudderElement.message.properties) {
+            eventValue = rudderElement.message.properties.value
+              ? rudderElement.message.properties.value
+              : rudderElement.message.properties.revenue;
+          }
+
+          var payload = {
+            eventCategory: eventCategory || "All",
+            eventAction: eventAction,
+            eventLabel: eventLabel,
+            eventValue: formatValue(eventValue),
+            // Allow users to override their nonInteraction integration setting for any single particluar event.
+            nonInteraction:
+              rudderElement.message.properties.nonInteraction !== undefined
+                ? !!rudderElement.message.properties.nonInteraction
+                : !!opts.nonInteraction,
+          };
+
+          var campaign = rudderElement.message.context.campaign;
+
+          if (campaign) {
+            if (campaign.name) payload.campaignName = campaign.name;
+            if (campaign.source) payload.campaignSource = campaign.source;
+            if (campaign.medium) payload.campaignMedium = campaign.medium;
+            if (campaign.content) payload.campaignContent = campaign.content;
+            if (campaign.term) payload.campaignKeyword = campaign.term;
+          }
+
+          payload = extend(
+            payload,
+            setCustomDimenionsAndMetrics(
+              rudderElement.message.properties,
+              this.inputs
+            )
+          );
+
+          ga("send", "event", payload);
+          logger.debug("in GoogleAnalyticsManager track");
+      }
+    } else {
+      var contextOpts; //need to implement
```

Comment:
this else is required ?
in the above default case of enhanced ecomm, we are doing the same logic.

which one should be kept?

Reply:
if the the enhanced e-commerce is true and non of the e-commerce event name doesn't match it will send normal events like if enhanced e-commerce is false. that is why I used else